cmd/kubeadm/app/cmd/token.go

```diff
@@ -277,22 +291,65 @@ func RunListTokens(out io.Writer, errW io.Writer, client clientset.Interface) er
 		return errors.Wrap(err, "failed to list bootstrap tokens")
 	}
 
-	w := tabwriter.NewWriter(out, 10, 4, 3, ' ', 0)
-	fmt.Fprintln(w, "TOKEN\tTTL\tEXPIRES\tUSAGES\tDESCRIPTION\tEXTRA GROUPS")
+	var tokens []*kubeadmapi.BootstrapToken
 	for _, secret := range secrets.Items {
-
 		// Get the BootstrapToken struct representation from the Secret object
 		token, err := kubeadmapi.BootstrapTokenFromSecret(&secret)
 		if err != nil {
 			fmt.Fprintf(errW, "%v", err)
 			continue
 		}
+		tokens = append(tokens, token)
+	}
+
+	if outputFormat == "text" {
+		w := tabwriter.NewWriter(out, 10, 4, 3, ' ', 0)
+		fmt.Fprintln(w, "TOKEN\tTTL\tEXPIRES\tUSAGES\tDESCRIPTION\tEXTRA GROUPS")
+		for _, token := range tokens {
+			// Get the human-friendly string representation for the token
+			humanFriendlyTokenOutput := humanReadableBootstrapToken(token)
+			fmt.Fprintln(w, humanFriendlyTokenOutput)
+		}
+		w.Flush()
+	} else if outputFormat == "yaml" || outputFormat == "json" {
+		clusterConfig, err := cmdutil.GetClusterConfig(kubeConfigFile)
+		if err != nil {
+			return errors.Wrapf(err, "failed to load cluster config from %s", kubeConfigFile)
+		}
+
+		publicKeyPins, err := cmdutil.GetCAPubKeyPins(clusterConfig)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get CA certs hashes from cluster config")
+		}
+
+		tokenList := BootstrapTokenList{
+			PublicKeyPins:        publicKeyPins,
+			ControlPlaneHostPort: strings.Replace(clusterConfig.Server, "https://", "", -1),
+			Tokens:               tokens,
+		}
+
+		items := struct {
```

Comment:
Not sure if you need `items` here. You can use `tokenList` directly, or is there a particular need to have everything to sit under a `bootstrapTokenList` in the output?

Reply:
yes, it's done exactly for having bootstrapTokenList in the output. This way the whole structure is more identified in my opinion.